storage/mysql/mysql_test.go

```diff
@@ -163,6 +164,93 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestGetTile(t *testing.T) {
+	ctx := context.Background()
+	s := newTestMySQLStorage(t, ctx)
+
+	awaiter := tessera.NewIntegrationAwaiter(ctx, s.ReadCheckpoint, 10*time.Millisecond)
+
+	treeSize := 258
+	var lastIndex uint64
+	for i := range treeSize {
+		idx, _, err := awaiter.Await(ctx, s.Add(ctx, tessera.NewEntry([]byte(fmt.Sprintf("TestGetTile %d", i)))))
+		if err != nil {
+			t.Fatalf("Failed to prep test with entry: %v", err)
+		}
+		if idx > lastIndex {
+			lastIndex = idx
+		}
+	}
+	if got, want := lastIndex, uint64(treeSize-1); got != want {
+		t.Fatalf("expected only newly created entries in database; tests are not hermetic (got %d, want %d)", got, want)
+	}
+
+	for _, test := range []struct {
+		name                   string
+		level, index, treeSize uint64
+		wantEntries            int
+		wantNotFound           bool
+	}{
+		{
+			name:  "really too small but that's ok",
```

Comment:
Trying to come up with a crisp description of what's being tested:
"treeSize << actual size, but all requested entries are present"?

Reply:
changed as discussed 1:1